### DISCUSSION TITLE
Running jobs in a container

### DISCUSSION THREAD

**Discussion by luigibrz** [actions/actions-runner-controller#3295] [Questions]
Hi, 

I've recently started testing ARC and I'm a bit new to this so apologies if this was asked before or is common sense for many.

I would like to understand what is the best configuration of ARC in order to run jobs in containers where the image to be used is defined in the GitHub Actions .yaml file and how it would be possible to cache those images in the cluster to speed up following runs.

EXAMPLE from doc page:
<img width="735" alt="image" src="https://github.com/actions/actions-runner-controller/assets/113340057/955cc85c-d9bd-4c84-a829-d91e4b352ce8">

https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container 

thanks a lot for any idea/example you can share to help!

**Comment by jonwest** [comment]
It looks like you're using the right configuration according to what has been defined within the documentation for running all steps for a job within a container.

In terms of caching images, this really depends on how you have ARC configured.  There will be a different kind of solution for DIND than there would be for Kubernetes runners.

For Docker-in-Docker, it depends on the capabilities of your underlying storage.  If you have performant RWX PVCs, then mounting one of those to all of your runners and configuring Docker to use that volume mount as the `dataDir` may be sufficient.  If not, then having a Kubernetes cronjob configured to use a RWO PVC and orchestrate a `docker pull` into that PVC, and having your cloud provider take snapshots of that PVC to be used as a clone source for a PVC within your runners to use as an image store could also work.
